doc/operator-capabilities.md

```diff
@@ -0,0 +1,193 @@
+# Operator Capability Levels
+
+Operators come in different maturity levels in regards to their lifecycle management capabilities for the application or workload they deliver. The capability models aims to provide guidance in terminology to express what features users can expect from an operator.
+
+![operator-capability-level](./images/operator-capability-level.png)
+
+Each capability level is associated with a certain set of management features the Operator offers around the managed workload. Operator that do not manage a workload and/or are delegating to off-clusters orchestration services would remain at Level 1. Capability levels are accumulating, i.e. Level 3 capabilities require all capabilities desired from Level 1 and 2.
+
+
+## Terminology
+
+**Operator** - the custom controller installed on a Kubernetes cluster
+
+**Operand** - the managed workload provided by the Operator as a service
+
+**Custom Resource (CR)** - an instance of the [`CustomResourceDefinition`](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/) the Operator ships that represents the Operand or an Operation on an Operand
+
+**Managed resources** - the Kubernetes objects or off-cluster services the Operator uses to constitute an Operand
+
+---
+
+## Level 1 - Basic Install
+
+The Operator offers the following basic features:
+
+**Installation of the workload**
+
+- Operator deploys an Operand or configures off-cluster resources
+- Operator waits for managed resources to reach a healthy state
+- Operator conveys readiness of application or managed resources to the user leveraging the `status` block of the Custom Resource
+
+**Example:** an Operator deploys a database by creating `Deployment`, `ServiceAccount`, `RoleBinding`, `ConfigMap`, `PersistentVolumeClaim` and `Secret` object, initializes an empty database schema and signals readiness of the database to accept queries.
+
+**Configuration of the workload**
+
+- Operator provides configuration via the `spec` section of the Custom Resource
+- Operator reconciles configuration and updates to it with the status of the managed resources
+
+**Example:** an Operator, managing a database, offers increasing the capacity of the database by resizing the underlying `PersistentVolumeClaim` based on changes the databases Custom Resource instance.
+
+**Guiding questions to determine Operator reaching Level 1**
+
+1. What installation configuration can be set in the CR?
+
+2. What additional CR installation configuration could still be added?
+
+3. Can you set operand configuration in the CR? If so, what configuration is supported for each operand?
+
```

Comment:
Should not all above be CRD and not CR?

Reply:
This looks at potential additions from the user perspective, so it would be CR. But I think the reader knows what this means when implemented.